databricks-sdk-java/src/main/java/com/databricks/sdk/core/commons/CommonsHttpClient.java

```diff
@@ -52,6 +52,7 @@ private RequestConfig makeRequestConfig() {
   private CloseableHttpClient makeClosableHttpClient() {
     return HttpClientBuilder.create()
         .setConnectionManager(connectionManager)
+        .disableRedirectHandling()
```

Comment:
Is this an Azure specific change? This will disable redirect handling for all cases. @mgyucht FYI

Reply:
Although the test cases aren't failing atleast, but let's wait for @mgyucht to confirm.

Reply:
Unit test cases probably won't cover this pathway. We can trigger integration tests here as well. However, I would prefer an option where you can enable/disable this on a per-request basis. If you look at the python SDK:
```python
        if self.is_azure:
            # Retrieve authorize endpoint to retrieve token endpoint after
            res = requests.get(f'{self.host}/oidc/oauth2/v2.0/authorize', allow_redirects=False)
            real_auth_url = res.headers.get('location')
            if not real_auth_url:
                return None
            return OidcEndpoints(authorization_endpoint=real_auth_url,
                                 token_endpoint=real_auth_url.replace('/authorize', '/token'))
```
We specifically disallow redirects for this one method.

HttpClient in apache commons supports a HttpRequestContext, which has a config controlling max # of redirects:
```java
  public CloseableHttpResponse execute(
            final HttpRoute route,
            final HttpRequestWrapper request,
            final HttpClientContext context,
            final HttpExecutionAware execAware) throws IOException, HttpException {
...
        final RequestConfig config = context.getRequestConfig();
        final int maxRedirects = config.getMaxRedirects() > 0 ? config.getMaxRedirects() : 50;
```
If we can plumb this through our Request interface, I think that would be best.

Reply:
Thanks for the idea, Miles. I have made the change